client/cli/src/main.rs

```diff
@@ -280,15 +289,38 @@ struct PayRequest {
 
 #[tokio::main]
 async fn main() {
-    let opts = Options::parse();
     tracing_subscriber::fmt()
         .with_env_filter(
             EnvFilter::try_from_default_env().unwrap_or_else(|_| EnvFilter::new("info")),
         )
         .with_writer(std::io::stderr)
         .init();
 
-    if let Command::JoinFederation { connect } = opts.command {
+    let cli = Cli::parse();
+
+    if let Command::VersionHash = cli.command {
+        println!(
+            "{}",
+            CliOutput::VersionHash {
+                hash: env!("GIT_HASH").to_string()
+            }
+        );
+        return;
+    };
+    //If we don't make the workdir optional future commands that might not need it will be
+    //bothered by that
+    let workdir = if let Some(workdir) = &cli.workdir {
```

Comment:
This part is not great.

I see couple of options/combinations of:

* Prepare a separate `#[derive(Parser)] struct CliNoWorkdir` that would be a subset of `Cli` containing only commands that don't need the workdir, and basically use `try_parse` to fall back between them.
* Parse `std::env::args_os` manually to handle this special case.
* Move workdir to be an argumento every subcommand that needs it. (kind of the cleanest way, but also most involved).
* See if https://docs.rs/clap/latest/clap/builder/struct.Arg.html#method.requires can be addded on all commands that require `workdir`.

Reply:
I think doing the other `Parser` might look clumsier but is cleaner than handling it with `std::env::args_os` directly. Good Idea

Reply:
yes "Any [Arg method](https://docs.rs/clap/latest/clap/builder/struct.Arg.html) can also be used as an attribute, see [Terminology](https://docs.rs/clap/latest/clap/_derive/index.html#terminology) for syntax." so it can be used in the derive API but I think I'll go with the two `Parsers`